File: ios/Exceptions.swift
import ExpoModulesCore

class NotSupportedException: Exception {
  override var reason: String {
    "Foundation Models requires iOS 26 or later"
  }
}

class SessionInvalidException: Exception {
  override var reason: String {
    "LLM session is invalid or has been destroyed"
  }
}

class StreamException: GenericException<String> {
  override var reason: String {
    "Stream error: \(param ?? "Unknown")"
  }
}

File: android/src/main/java/expo/modules/localllm/Exceptions.kt
package expo.modules.localllm

import expo.modules.kotlin.exception.CodedException

class NotSupportedException :
  CodedException("ERR_NOT_SUPPORTED", "On-device LLM is not available on this device", null)

class SessionInvalidException :
  CodedException("ERR_SESSION_INVALID", "LLM session is invalid or has been destroyed", null)

class StreamException(message: String) :
  CodedException("ERR_STREAM", message, null)

class ToolNotSupportedException :
  CodedException("ERR_TOOL_NOT_SUPPORTED", "Tool calling is not supported on Android yet", null)
